src/CoaddBoundedField.cc

```diff
@@ -239,6 +239,18 @@ PTR(afw::math::BoundedField) CoaddBoundedField::operator*(double const scale) co
     throw LSST_EXCEPT(pex::exceptions::NotFoundError, "Scaling of CoaddBoundedField is not implemented");
 }
 
+bool CoaddBoundedField::operator==(BoundedField const& rhs) const {
+    try {
+        auto& rhsCasted = dynamic_cast<CoaddBoundedField const&>(rhs);
+        return (getBBox() == rhsCasted.getBBox()) && (_default == rhsCasted._default) &&
+               ((*_coaddWcs) == (*rhsCasted._coaddWcs)) && (_elements == rhsCasted._elements);
+    }
+    catch (std::bad_cast&) {
+        return false;
+    }
```

Comment:
Thanks for the suggestion. I've implemented this here, and done the same for ChebyshevBoundedField.